packages/graphql-transformer-core/src/util/fileUtils.ts

```diff
@@ -0,0 +1,183 @@
+function promisify<A, O>(fn: (arg: A, cb: (err: Error, data: O) => void) => void, a: A): Promise<O> {
+    return new Promise((res, rej) => {
+        fn(a, (err, d) => {
+            err ? rej(err) : res(d)
+        })
+    })
+}
+
+/**
+ * Helpers
+ */
+
+export async function deleteDirectory(directory: string): Promise<void> {
+    const pathExists = await exists(directory);
+    if (!pathExists) {
+        return;
+    }
+    await emptyDirectory(directory);
+    await rmdir(directory);
+}
+
+export async function emptyDirectory(directory: string) {
+    const pathExists = await exists(directory);
+    if (!pathExists) {
+        return;
+    }
+    const dirStats = await lstat(directory);
+    if (!dirStats.isDirectory()) {
+        return;
+    }
+    const files = await readDir(directory);
+    for (const fileName of files) {
+        const fullPath = path.join(directory, fileName);
+        const stats = await lstat(fullPath);
+        if (stats.isDirectory()) {
+            await deleteDirectory(fullPath);
+        } else if (stats.isFile()) {
+            await unlink(fullPath);
+        }
+    }
+}
+
+/**
+ * Make if not exists.
+ */
+export async function mkdirIfNone(dir: string) {
+    const pathExists = await exists(dir);
+    if (!pathExists) {
+        await mkdir(dir)
+    }
+}
+
+export async function writeToPath(directory: string, obj: any): Promise<void> {
+    if (Array.isArray(obj)) {
+        await mkdirIfNone(directory);
+        for (let i = 0; i < obj.length; i++) {
+            const newDir = path.join(directory, `${i}`);
+            await writeToPath(newDir, obj[i]);
+        }
+    } else if (typeof obj === 'object') {
+        await mkdirIfNone(directory);
+        for (const key of Object.keys(obj)) {
+            const newDir = path.join(directory, key);
+            await writeToPath(newDir, obj[key])
+        }
+    } else if (typeof obj === 'string') {
+        fs.writeFileSync(directory, obj)
+    }
+}
+
+/**
+ * Recursively read the contents of a directory into an object.
+ * @param directory The directory to read.
+ */
+export async function readFromPath(directory: string): Promise<any> {
+    const pathExists = await exists(directory);
+    if (!pathExists) {
+        return;
+    }
+    const dirStats = await lstat(directory);
+    if (!dirStats.isDirectory()) {
+        const buf = await readFile(directory);
+        return buf.toString();
+    }
+    const files = await readDir(directory);
+    const accum = {};
+    for (const fileName of files) {
+        const fullPath = path.join(directory, fileName);
+        const value = await readFromPath(fullPath);
+        accum[fileName] = value;
+    }
+    return accum;
+}
+
+/**
+ * Recursively clear the contents at some path. Might be a file or directory.
+ * @param clearPath The path to clear.
+ */
+export async function clearAtPath(clearPath: string) {
```

Comment:
`remove` / `removeSync` from `fs-extra` should do what we are doing here

Reply:
Replaced